lib/pages/chat/chat.dart

```diff
@@ -1547,10 +1497,69 @@ class ChatController extends State<Chat>
     if (popResult is Event) {
       scrollToEventIdAndHighlight(popResult.eventId);
     } else if (popResult is List<Event?>) {
-      pinnedEventsController.handlePopBack(popResult);
+      pinnedEventsController.handlePopBack(
+        popResult: popResult,
+        client: client,
+      );
     }
   }
 
+  void _initializePinnedEvents() {
+    if (roomId == null) return;
+    pinnedEventsController.getPinnedMessageAction(
+      client: client,
+      roomId: roomId!,
+      isInitial: true,
+    );
+  }
+
+  @override
+  void initState() {
+    _initializePinnedEvents();
```

Comment:
should we put it after `_tryLoadTimeline()` ?

Reply:
Not necessary, It will run in parallel

Reply:
so we need to call `getTimeline()` 2 times ?

Reply:
👍🏻

Reply:
Why not use the first solution, we don't have to call it 2 times

Reply:
Because first solution  impact on the displaying of message to chat list

Reply:
The second thing is that `getTimeLine` will get data from the cache, calling it twice does not affect performance too much.

Reply:
okay, trade off